pyk/src/pyk/kcfg/tui.py

```diff
@@ -173,8 +173,9 @@ def on_click(self, click: Click) -> None:
 
 
 class NodeView(Widget):
-    _kprint: KPrint
+    _definition: KDefinition
     _custom_view: Callable[[KCFGElem], Iterable[str]] | None
+    _cterm_show: CTermShow
```

Comment:
Target for future refactorings: abstract interfaces should be extracted.

Reply:
I agree. I'm playing with that a bit on my machine, but I think we need a bit of a planning session around that. Currently, this change is an improvement that gets us some benefit and a slighlty cleaner interface, and I have a few more improvements planned, but I need to propagate it downstream before going too far I think.